src/Clinics/Domain/Actions/StoreClinicAction.php

```
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lightit\Clinics\Domain\Actions;
+
+use Lightit\Clinics\Domain\DataTransferObjects\ClinicDto;
+use Lightit\Clinics\Domain\Models\Clinic;
+
+class StoreClinicAction
+{
+    public function execute(ClinicDto $clinicDto): Clinic
+    {
+        $clinic = new Clinic();
+
+        $clinic->name = $clinicDto->name;
+        $clinic->address = $clinicDto->address;
+
+        $clinic->save();
```

saveOrFail() to prevent silent failure

applies to all!